### PR TITLE
Feature same module inject

### PR DESCRIPTION
**Pull request by taobaorun**
1.componentinfo add more details for components endpoint.
2.reference check support skip by interface & uniqueId

### CODE REVIEW

**Review by HzjNeverStop — CHANGES_REQUESTED** [review]

File: sofa-boot-project/sofa-boot-core/runtime-sofa-boot/src/main/java/com/alipay/sofa/runtime/spi/component/ComponentDefinitionInfo.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ * @version : ComponentDefinitionInfo.java, v 0.1 2023年03月09日 14:38 yuanxuan Exp $
+ */
+public class ComponentDefinitionInfo {
+
+    public static final String SOURCE = "source";
+
+    /**
+     * annotation or xml
+     */
+    private SourceType         sourceType;
```
[the remark this answers is not in the document]
Comment (taobaorun) [reply sofa-boot-project/sofa-boot-core/runtime-sofa-boot/src/main/java/com/alipay/sofa/runtime/spi/component/ComponentDefinitionInfo.java:32]: ok

File: sofa-boot-project/sofa-boot-core/runtime-sofa-boot/src/main/java/com/alipay/sofa/runtime/SofaRuntimeProperties.java [hunk 2]
```diff
@@ -26,19 +26,22 @@
...
+    private static final ConcurrentHashMap<ClassLoader, Boolean>  skipJvmReferenceHealthCheckMap      = new ConcurrentHashMap<>();
+    private static final ConcurrentHashMap<ClassLoader, Boolean>  skipExtensionHealthCheckMap         = new ConcurrentHashMap<>();
+    private static final ConcurrentHashMap<ClassLoader, Boolean>  disableJvmFirstMap                  = new ConcurrentHashMap<>();
+    private static final ConcurrentHashMap<ClassLoader, Boolean>  skipJvmSerializeMap                 = new ConcurrentHashMap<>();
+    private static final ConcurrentHashMap<ClassLoader, Boolean>  extensionFailureInsulatingMap       = new ConcurrentHashMap<>();
+    private static final ConcurrentHashMap<ClassLoader, Boolean>  manualReadinessCallbackMap          = new ConcurrentHashMap<>();
+    private static final ConcurrentHashMap<ClassLoader, Boolean>  skipAllComponentShutdownMap         = new ConcurrentHashMap<>();
+    private static final ConcurrentHashMap<ClassLoader, Boolean>  skipCommonComponentShutdownMap      = new ConcurrentHashMap<>();
+
+    private static final ConcurrentHashMap<ClassLoader, String[]> skipJvmReferenceHealthCheckArrayMap = new ConcurrentHashMap<>();
```
[the remark this answers is not in the document]
Comment (taobaorun) [reply sofa-boot-project/sofa-boot-core/runtime-sofa-boot/src/main/java/com/alipay/sofa/runtime/SofaRuntimeProperties.java:38]: ok

File: sofa-boot-project/sofa-boot-core/runtime-sofa-boot/src/main/java/com/alipay/sofa/runtime/configure/SofaRuntimeConfigurationProperties.java [hunk 3]
```diff
@@ -141,4 +141,14 @@ public boolean isServiceNameWithBeanId() {
         return SofaRuntimeProperties.isServiceNameWithBeanId();
     }
 
+    public void setSkipJvmReferenceHealthCheckArray(String[] skipJvmReferenceHealthCheckArray) {
```
[the remark this answers is not in the document]
Comment (taobaorun) [reply sofa-boot-project/sofa-boot-core/runtime-sofa-boot/src/main/java/com/alipay/sofa/runtime/configure/SofaRuntimeConfigurationProperties.java:144]: ok

File: sofa-boot-project/sofa-boot-core/runtime-sofa-boot/src/main/java/com/alipay/sofa/runtime/service/component/ReferenceComponent.java [hunk 4]
```diff
@@ -56,6 +58,8 @@ public class ReferenceComponent extends AbstractComponent {
     private Reference                 reference;
     private CountDownLatch            latch                    = new CountDownLatch(1);
 
+    private Map<String, Property>     properties               = new ConcurrentHashMap<>();
```
[the remark this answers is not in the document]
Comment (taobaorun) [reply sofa-boot-project/sofa-boot-core/runtime-sofa-boot/src/main/java/com/alipay/sofa/runtime/service/component/ReferenceComponent.java:61]: ok

File: sofa-boot-project/sofa-boot-core/runtime-sofa-boot/src/main/java/com/alipay/sofa/runtime/service/component/ReferenceComponent.java [hunk 5]
```diff
@@ -100,11 +104,36 @@ public HealthResult isHealthy() {
         // check reference has a corresponding service
         if (!SofaRuntimeProperties.isSkipJvmReferenceHealthCheck(sofaRuntimeContext)
             && jvmBinding != null) {
-            Object serviceTarget = getServiceTarget();
-            if (serviceTarget == null && !jvmBinding.hasBackupProxy()) {
-                jvmBindingHealthResult.setHealthy(false);
-                jvmBindingHealthResult.setHealthReport("can not find corresponding jvm service");
+            //skip check reference for the specified interface with unique id
+            String[] skipCheckArray = SofaRuntimeProperties
+                .getSkipJvmReferenceHealthCheckArray(Thread.currentThread().getContextClassLoader());
```
[the remark this answers is not in the document]
Comment (taobaorun) [reply sofa-boot-project/sofa-boot-core/runtime-sofa-boot/src/main/java/com/alipay/sofa/runtime/service/component/ReferenceComponent.java:109]: ok

File: sofa-boot-project/sofa-boot-autoconfigure/src/test/java/com/alipay/sofa/autoconfigure/test/runtime/SofaRuntimePropertiesTest.java [hunk 6]
```diff
@@ -27,21 +29,20 @@
...
-@TestPropertySource(properties = { "com.alipay.sofa.boot.disableJvmFirst=true",
+@TestPropertySource(properties = {
+                                  "com.alipay.sofa.boot.disableJvmFirst=true",
                                   "com.alipay.sofa.boot.skipJvmReferenceHealthCheck=true",
                                   "com.alipay.sofa.boot.skipJvmSerialize=true",
                                   "com.alipay.sofa.boot.skipExtensionHealthCheck=true",
                                   "com.alipay.sofa.boot.extensionFailureInsulating=true",
-                                  "com.alipay.sofa.boot.serviceInterfaceTypeCheck=true" })
+                                  "com.alipay.sofa.boot.serviceInterfaceTypeCheck=true",
+                                  "com.alipay.sofa.boot.skipJvmReferenceHealthCheckArray=com.alipay.sofa.isle.sample.facade.SampleJvmService:annotationImpl,com.alipay.sofa.isle.sample.facade.SampleJvmService" })
```
[the remark this answers is not in the document]
Comment (taobaorun) [reply sofa-boot-project/sofa-boot-autoconfigure/src/test/java/com/alipay/sofa/autoconfigure/test/runtime/SofaRuntimePropertiesTest.java:45]: done

File: sofa-boot-project/sofa-boot-actuator/src/main/java/com/alipay/sofa/boot/actuator/components/SofaBootComponentsEndPoint.java [hunk 7]
```diff
@@ -56,8 +59,12 @@ public ApplicationComponents components() {
             Collection<ComponentInfo> componentInfos = componentManager.getComponentInfosByType(componentType);
             Collection<ComponentDisplayInfo> componentDisplayInfos = componentInfos.stream()
                     .map(componentInfo -> {
-                        String applicationId = componentInfo.getApplicationContext() == null ? "null" : componentInfo.getApplicationContext().getId();
-                        return new ComponentDisplayInfo(componentInfo.getName().getName(), applicationId);
+                        String applicationId = componentInfo.getApplicationContext() == null ? "null"
```
Comment (HzjNeverStop) [inline sofa-boot-project/sofa-boot-actuator/src/main/java/com/alipay/sofa/boot/actuator/components/SofaBootComponentsEndPoint.java:62]: 补充下相关的测试用例，验证getProperties的读取正确？

**Review by HzjNeverStop — APPROVED** [review]
LGTM

**Pull request merged**